pkg/controllers/endpointaccessible/endpoint_accessible_controller.go

```diff
@@ -183,10 +183,17 @@ func (c *endpointAccessibleController) sync(ctx context.Context, syncCtx factory
 			req.WithContext(reqCtx)
 
 			// we don't really care  if anyone lies to us. We aren't sending important data.
-			insecureTransport := &http.Transport{
-				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			client := &http.Client{
+				Transport: &http.Transport{
+					Proxy: http.ProxyFromEnvironment,
+					DialContext: (&net.Dialer{
+						Timeout: 5 * time.Second,
+					}).DialContext,
```

Comment:
It feels like it would be simpler to just set the `Timeout` attribute of the `http.Client` struct, is there any reason not to?

Reply:
@stlaz I had only tcp timeout in my mind when I was writing this. But, I guess client `Timeout` also works in this case and would be simpler too. Update the code. Please take a look.